### PR TITLE
Add `notebook`/`pad`[`-*`] icons & optimise/rename `*-edit` icons

### PR DESCRIPTION
**Pull request by danielbayley**
Moved directly from https://github.com/lucide-icons/lucide/pull/1352#pullrequestreview-1473797256 …

`notepad-text-dashed` is equivalent to https://lucide.dev/icons/book-template (to be renamed to `book-dashed` in #1352), and covers `notebook-template` in [codicons](https://code.visualstudio.com/api/references/icons-in-labels#icon-listing).

### CODE REVIEW

**Review by jguddas — APPROVED** [review]

File: icons/notebook-dashed.svg
Comment (karsa-mistmere) [inline icons/notebook-dashed.svg:1]: This feels a bit more like some kind of transformer circuit. 🤔

File: icons/notebook-pencil.svg
Comment (karsa-mistmere) [inline icons/notebook-pencil.svg:1]: Something like this (using the pencil shape from `file-/clipboard-/folder-edit`) would feel more trivial to me:
image

**Review by karsa-mistmere — CHANGES_REQUESTED** [review]

File: icons/notebook-pencil.svg
Comment (karsa-mistmere) [reply icons/notebook-pencil.svg:1]: I see that you've renamed this one, on the other hand, maybe if `edit` icons are renamed `pen` and `pencil` in #1286, `*-edit` icons should follow suit as `*-pen`?... 🤔

@ericfennis, what do you think?

Comment (danielbayley) [reply icons/notebook-pencil.svg:1]: > `*-edit` icons should follow suit as `*-pen`?

Yeah, I also did think that…

**Review by jguddas — APPROVED** [review]

**Pull request merged**